packages/streaming-image-volume-loader/src/BaseStreamingImageVolume.ts

```diff
@@ -243,280 +460,129 @@ export default class BaseStreamingImageVolume extends ImageVolume {
-      }
+    if (scalingParameters.modality === 'PT') {
+      const suvFactor = metaData.get('scalingModule', imageId);
 
-      // if it is not a cached image or volume
-      if (
-        !cachedImage?.image &&
-        !(cachedVolume && cachedVolume.volume !== this)
-      ) {
-        return updateTextureAndTriggerEvents(this, imageIdIndex, imageId);
+      if (suvFactor) {
+        this._addScalingToVolume(suvFactor);
+        scalingParameters.suvbw = suvFactor.suvbw;
       }
+    }
 
-      // it is either cachedImage or cachedVolume
-      const isFromImageCache = !!cachedImage;
-
-      const cachedImageOrVolume = cachedImage || cachedVolume.volume;
+    const isSlopeAndInterceptNumbers =
+      typeof scalingParameters.rescaleSlope === 'number' &&
+      typeof scalingParameters.rescaleIntercept === 'number';
+
+    /**
+     * So this is has limitation right now, but we need to somehow indicate
+     * whether the volume has been scaled with the scaling parameters or not.
+     * However, each slice can have different scaling parameters but it is rare
+     * that rescale slope and intercept be unknown for one slice and known for
+     * another. So we can just check the first slice and assume that the rest
+     * of the slices have the same scaling parameters. Basically it is important
+     * that these two are numbers and that means the volume has been scaled (
+     * we do that automatically in the loader). For the suvbw, we need to
+     * somehow indicate whether the PT image has been corrected with suvbw or
+     * not, which we store it in the this.scaling.PT.suvbw.
+     */
+    this.isPreScaled = isSlopeAndInterceptNumbers;
 
-      this.handleImageComingFromCache(
-        cachedImageOrVolume,
-        isFromImageCache,
+    return {
+      // WADO Image Loader
+      targetBuffer: {
+        // keeping this in the options means a large empty volume array buffer
+        // will be transferred to the worker. This is undesirable for streaming
+        // volume without shared array buffer because the target is now an empty
+        // 300-500MB volume array buffer. Instead the volume should be progressively
+        // set in the main thread.
+        arrayBuffer:
+          arrayBuffer instanceof ArrayBuffer ? undefined : arrayBuffer,
+        offset: imageIdIndex * lengthInBytes,
+        length,
+        type,
+        rows,
+        columns,
+      },
+      skipCreateImage: true,
+      preScale: {
+        enabled: true,
+        // we need to pass in the scalingParameters here, since the streaming
+        // volume loader doesn't go through the createImage phase in the loader,
+        // and therefore doesn't have the scalingParameters
         scalingParameters,
-        scalarData,
-        frameIndex,
-        arrayBuffer,
-        updateTextureAndTriggerEvents,
-        imageIdIndex,
-        imageId,
-        errorCallback
-      );
+      },
+      transferSyntaxUID,
+      loader: imageLoader.loadImage,
```

Comment:
I don't see anywhere in the code where we use this `loader`. Why do we have it here? If it's as simple as `imageLoader.loadImage`, then one can import it.

Reply:
It gets used in the progressive loader to distinguish between loadImage and loadAndCacheImage.  It was pre-existing, so I figured I would just use it as I wanted the function to be in the options as it is really about how the load occurs.